pkg/proxy/iptables/proxier.go

```diff
@@ -656,6 +660,9 @@ func (proxier *Proxier) OnNodeDelete(node *v1.Node) {
 			"eventNode", node.Name, "currentNode", proxier.hostname)
 		return
 	}
+
+	proxier.healthzServer.SyncNode(node)
```

Comment:
it is better if you add your own handler, per example `NodeHealthzHandler`and register it during the kube-proxy initialization, adding the feature gate on registration, so you don't have to plumb it in all the proxies and doesn't gate executed despite is feature gated

See https://github.com/kubernetes/kubernetes/pull/111344 for reference

Reply:
I can't follow, but I'm a bit slow these days, so I may be missing something

> n this case we always want to compute/react to the Node event, but consider/not consider it depending on the feature gate.

why do you want to compute it if you never going to use it, you always have to restart ... the feature gate just does that, not execute code that is under feature gate

Reply:
Feature gates can't (today) change live - they always require a restart.  That said, this plumbing is minor - I can go either way.  Antonio has more context on current best-practice :)